bigquery/routine_integration_test.go

```diff
@@ -88,6 +92,133 @@ func TestIntegration_RoutineJSUDF(t *testing.T) {
 	}
 }
 
+func TestIntegration_RoutineRemoteUDF(t *testing.T) {
+	if client == nil {
+		t.Skip("Integration tests skipped")
+	}
+	ctx := context.Background()
+
+	parentLocation := fmt.Sprintf("projects/%s/locations/%s", dataset.ProjectID, "us-central1")
+	routineID := routineIDs.New()
+	routine := dataset.Routine(routineID)
+
+	functionName := fmt.Sprintf("udf-func-%s", strings.ReplaceAll(routineID, "_", "-"))
+	cleanupFunction, uri, err := createCloudFunction(ctx, t, parentLocation, functionName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanupFunction()
+
+	parentLocation = fmt.Sprintf("projects/%s/locations/%s", dataset.ProjectID, "us")
+	connectionName := fmt.Sprintf("udf_conn%s", routineID)
+	cleanupConnection, connectionID, err := createConnection(ctx, t, parentLocation, connectionName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanupConnection()
+
+	remoteOpts := &RemoteFunctionOptions{
+		Endpoint:           uri,
+		Connection:         connectionID,
+		MaxBatchingRows:    50,
+		UserDefinedContext: map[string]string{"foo": "bar"},
+	}
+	meta := &RoutineMetadata{
+		RemoteFunctionOptions: remoteOpts,
+		Description:           "defines a remote function",
+		Type:                  ScalarFunctionRoutine,
+		ReturnType: &StandardSQLDataType{
+			TypeKind: "STRING",
+		},
+	}
+	if err := routine.Create(ctx, meta); err != nil {
+		t.Fatalf("routine.Create: %v", err)
+	}
+
+	gotMeta, err := routine.Metadata(ctx)
+	if err != nil {
+		t.Fatalf("routine.Metadata: %v", err)
+	}
+
+	if diff := testutil.Diff(gotMeta.RemoteFunctionOptions, remoteOpts); diff != "" {
+		t.Fatalf("RemoteFunctionOptions: -got, +want:\n%s", diff)
+	}
+}
+
+func createCloudFunction(ctx context.Context, t *testing.T, parent, functionName string) (func(), string, error) {
```

Comment:
given the return style you end up using in this function, maybe just using named return args makes sense?